File: frontend/assets/js/portfolio-profile.js
export const PORTFOLIO_PROFILE = {
  name: "NICOLAS KEVEN LOPES DE HOLANDA",
  title: "Desenvolvedor Web",
  summary:
    "Ola, eu sou Nicolas Keven Lopes de Holanda, desenvolvedor web e estudante de Gestao da Tecnologia da Informacao. Tenho conhecimentos em Engenharia de Software e em diferentes linguagens e tecnologias de programacao, incluindo JavaScript, HTML, C++, C# e Python. Atualmente, curso Gestao da Tecnologia da Informacao e tambem possuo curso tecnico de linguagem de programacao Python e curso de Informatica. Tenho interesse em desenvolvimento de software, tecnologia, programacao e criacao de solucoes digitais, buscando evoluir continuamente meus conhecimentos e minhas habilidades na area de tecnologia.",
  education: ["Faculdade de Gestao da Tecnologia da Informacao"],
  courses: ["Curso tecnico de linguagem de programacao Python", "Curso de Informatica"],
  skillsByCategory: [
    {
      category: "Programacao",
      items: [
        { name: "JavaScript", level: "Conhecimento" },
        { name: "HTML", level: "Conhecimento" },
        { name: "C++", level: "Conhecimento" },
        { name: "C#", level: "Conhecimento" },
        { name: "Python", level: "Conhecimento" },
      ],
    },
    {
      category: "Engenharia de Software",
      items: [{ name: "Engenharia de Software", level: "Conhecimento" }],
    },
    {
      category: "Tecnologia",
      items: [
        { name: "Gestao da Tecnologia da Informacao", level: "Estudando" },
        { name: "Informatica", level: "Conhecimento" },
      ],
    },
  ],
  projects: [
    {
      name: "Orion PWA",
      description: "Assistente pessoal com chat, memoria, voz, arquivos e pesquisa web.",
      technologies: ["Python", "FastAPI", "JavaScript", "PWA", "SQLite"],
      category: "Assistente de IA",
      image: "Nao informado",
      link: "Nao informado",
      github: "Nao informado",
      status: "Ativo",
    },
  ],
};

export function buildPortfolioPresentationScript() {
  return [
    "Ola! Este e o portfolio de Nicolas Keven Lopes de Holanda.",
    "Nicolas e desenvolvedor web e possui conhecimentos em Engenharia de Software e em linguagens como JavaScript, HTML, C++, C# e Python.",
    "Atualmente, cursa Gestao da Tecnologia da Informacao e tambem possui curso tecnico de linguagem de programacao Python e curso de Informatica.",
    "Neste portfolio, voce pode conhecer mais sobre formacao, conhecimentos, habilidades e evolucao na area de tecnologia.",
  ].join(" ");
}

export function renderPortfolioSmartGrid(container, profile = PORTFOLIO_PROFILE) {
  if (!container) {
    return;
  }
  container.replaceChildren();

  profile.skillsByCategory.forEach((category) => {
    const card = document.createElement("article");
    card.className = "portfolio-smart-card";

    const heading = document.createElement("h3");
    heading.textContent = category.category;
    card.appendChild(heading);

    const list = document.createElement("ul");
    list.className = "portfolio-smart-list";

    category.items.forEach((item) => {
      const entry = document.createElement("li");
      entry.className = "portfolio-smart-item";

      const label = document.createElement("span");
      label.className = "portfolio-skill-name";
      label.textContent = item.name;

      const badge = document.createElement("strong");
      badge.className = "portfolio-skill-badge";
      badge.dataset.level = normalizeLevel(item.level);
      badge.textContent = item.level;

      entry.append(label, badge);
      list.appendChild(entry);
    });

    card.appendChild(list);
    container.appendChild(card);
  });
}

export function renderPortfolioEvolution(container, profile = PORTFOLIO_PROFILE) {
  if (!container) {
    return;
  }
  container.replaceChildren();

  const items = profile.skillsByCategory.flatMap((category) => category.items);
  items.forEach((item) => {
    const row = document.createElement("div");
    row.className = "portfolio-evolution-row";

    const name = document.createElement("span");
    name.className = "portfolio-evolution-name";
    name.textContent = item.name;

    const level = document.createElement("span");
    level.className = "portfolio-evolution-level";
    level.dataset.level = normalizeLevel(item.level);
    level.textContent = item.level;

    row.append(name, level);
    container.appendChild(row);
  });
}

export function renderPortfolioProjects(container, profile = PORTFOLIO_PROFILE) {
  if (!container) {
    return;
  }
  container.replaceChildren();

  profile.projects.forEach((project) => {
    const card = document.createElement("article");
    card.className = "portfolio-card";
    card.setAttribute("data-portfolio-card", "");

    const title = document.createElement("h3");
    title.textContent = project.name;

    const description = document.createElement("p");
    description.textContent = project.description;

    const meta = document.createElement("p");
    meta.className = "portfolio-project-meta";
    meta.textContent = `Categoria: ${project.category} | Tecnologias: ${project.technologies.join(", ")} | Status: ${project.status}`;

    const links = document.createElement("p");
    links.className = "portfolio-project-links";
    links.textContent = `Link: ${project.link} | GitHub: ${project.github}`;

    card.append(title, description, meta, links);
    container.appendChild(card);
  });
}

function normalizeLevel(level) {
  const normalized = String(level || "").toLowerCase();
  if (normalized.includes("desenvolvimento")) {
    return "developing";
  }
  if (normalized.includes("estudando")) {
    return "studying";
  }
  if (normalized.includes("familiar")) {
    return "familiar";
  }
  return "knowledge";
}

File: app/brain/portfolio_profile.py
from dataclasses import dataclass

from app.brain.orion_intents import normalize_text


@dataclass(frozen=True)
class PortfolioProfile:
    name: str
    title: str
    summary: str
    education: list[str]
    courses: list[str]
    skills: list[str]
    interests: list[str]


PROFILE = PortfolioProfile(
    name="NICOLAS KEVEN LOPES DE HOLANDA",
    title="Desenvolvedor Web",
    summary=(
        "Ola, eu sou Nicolas Keven Lopes de Holanda, desenvolvedor web e estudante de Gestao da Tecnologia da "
        "Informacao. Tenho conhecimentos em Engenharia de Software e em diferentes linguagens e tecnologias de "
        "programacao, incluindo JavaScript, HTML, C++, C# e Python. Atualmente, curso Gestao da Tecnologia da "
        "Informacao e tambem possuo curso tecnico de linguagem de programacao Python e curso de Informatica. "
        "Tenho interesse em desenvolvimento de software, tecnologia, programacao e criacao de solucoes digitais, "
        "buscando evoluir continuamente meus conhecimentos e minhas habilidades na area de tecnologia."
    ),
    education=["Faculdade de Gestao da Tecnologia da Informacao"],
    courses=[
        "Curso tecnico de linguagem de programacao Python",
        "Curso de Informatica",
    ],
    skills=[
        "JavaScript",
        "HTML",
        "C++",
        "C#",
        "Python",
        "Engenharia de Software",
        "Gestao da Tecnologia da Informacao",
        "Informatica",
    ],
    interests=[
        "desenvolvimento de software",
        "tecnologia",
        "programacao",
        "criacao de solucoes digitais",
    ],
)


def portfolio_voice_presentation() -> str:
    return (
        "Ola! Este e o portfolio de Nicolas Keven Lopes de Holanda. "
        "Nicolas e desenvolvedor web e possui conhecimentos em Engenharia de Software e em linguagens como "
        "JavaScript, HTML, C++, C# e Python. "
        "Atualmente, cursa Gestao da Tecnologia da Informacao e tambem possui curso tecnico de linguagem de "
        "programacao Python e curso de Informatica. "
        "Neste portfolio, voce pode conhecer mais sobre formacao, conhecimentos, habilidades e evolucao na area "
        "de tecnologia."
    )


def answer_portfolio_question(user_text: str) -> str | None:
    normalized = normalize_text(user_text)

    if any(term in normalized for term in {"quem e nicolas", "conte mais sobre nicolas", "quem e ele"}):
        return f"{PROFILE.name} e {PROFILE.title}. {PROFILE.summary}"
    if any(term in normalized for term in {"formacao", "formacao dele", "o que ele estuda"}):
        return f"Formacao atual: {PROFILE.education[0]}."
    if any(term in normalized for term in {"curso", "cursos", "quais sao os cursos"}):
        return "Cursos cadastrados: " + ", ".join(PROFILE.courses) + "."
    if any(term in normalized for term in {"habilidades", "linguagens", "conhecimentos", "tecnologias"}):
        return "Conhecimentos cadastrados no portfolio: " + ", ".join(PROFILE.skills) + "."
    if any(term in normalized for term in {"sobre mim", "resumo profissional"}):
        return PROFILE.summary

    if "portfolio" in normalized or "nicolas" in normalized:
        return (
            f"No portfolio, {PROFILE.name} e apresentado como {PROFILE.title}, com formacao em "
            "Gestao da Tecnologia da Informacao, cursos de Python e Informatica, e conhecimentos tecnicos "
            "cadastrados de forma verificavel."
        )

    return None
